tests/test_priority_samples_process.py

```diff
@@ -0,0 +1,379 @@
+
+        for doc in self.expected_dart_samples:
+            self.mock_get_well_index.assert_any_call(doc[FIELD_COORDINATE])
+
+        # wells created in dart
+        assert self.mock_set_well_props.call_count == num_wells
+
+        # Wells created from plate
+        if num_wells > 0:
+            for barcode in self.expected_dart_plates:
+                if self.plate_status[barcode] == DART_STATE_PENDING:
+                    self.mock_set_well_props.assert_any_call(
+                        self.mock_conn().cursor(), barcode, self.test_well_props, self.test_well_index
+                    )
+
+    def test_commits_changes_to_dart(self, mongo_database, config, with_different_scenarios):
+        _, mongo_database = mongo_database
+
+        update_priority_samples(mongo_database, config, True)
+
+        # commits changes
+        self.mock_conn().cursor().rollback.assert_not_called()
+
+        # 1 commit per pending plate
+        assert self.mock_conn().cursor().commit.call_count == len(self.expected_dart_plates)
+        self.mock_conn().close.assert_called_once()
+
+    def test_adding_plate_and_wells_to_dart_fails_with_expection(self, mongo_database, config):
+        _, mongo_database = mongo_database
+
+        with patch("crawler.priority_samples_process.add_dart_well_properties", side_effect=Exception("Boom!")):
+            update_priority_samples(mongo_database, config, True)
+
+            assert logging_collection.get_count_of_all_errors_and_criticals() >= 1
+            assert logging_collection.aggregator_types["TYPE 33"].count_errors == 1
+
+    def test_adding_plate_and_wells_insert_failed(self, mongo_database, config):
+        _, mongo_database = mongo_database
+
+        with patch("crawler.priority_samples_process.create_dart_sql_server_conn") as mocked_conn:
+            mocked_conn().cursor.side_effect = Exception("Boom!!")
+            update_priority_samples(mongo_database, config, True)
+
+            assert logging_collection.get_count_of_all_errors_and_criticals() >= 1
+            assert logging_collection.aggregator_types["TYPE 30"].count_errors == 1
+
+    def test_dart_sql_server_cannot_connect(self, config, mongo_database):
+        _, mongo_database = mongo_database
+
+        with patch("crawler.priority_samples_process.create_dart_sql_server_conn") as mock_conn:
+            mock_conn.return_value = None
+
+            update_priority_samples(mongo_database, config, True)
+
+            assert logging_collection.get_count_of_all_errors_and_criticals() >= 1
+            assert logging_collection.aggregator_types["TYPE 31"].count_errors == 1
+
+    def test_centre_config_for_samples(self, config):
+        result = centre_config_for_samples(config, [{FIELD_SOURCE: "Test Centre"}])
+        assert result["name"] == "Test Centre"
+        assert result["prefix"] == "TEST"
+        assert result["lab_id_default"] == "TE"
+
+
+# We have priority samples that have not been received yet (not in mongodb)
```

Comment:
Are these some future tests or WIP?

Reply:
Tests we wrote at the start when deciding what tests to write, they should be either redundant or covered now in above tests